Test.hs

```diff
@@ -22,72 +21,71 @@ main = do
   putStrLn "VHDL:"
   testVHDL
   putStrLn "\n------All Tests Passed------"
-  where
 
 testShowInstance :: IO ()
-testShowInstance = case a == (show b) of
-  True -> putStrLn "-Pangraph show instance passed."
-  False -> error $ "!Show instance failed test: " ++ show b ++ "!=" ++ a
-  where
-    a = fst showInstanceTestCase
-    b = snd showInstanceTestCase
-    showInstanceTestCase :: (String, P.Pangraph)
-    showInstanceTestCase = ("Pangraph [Node [Attribute (\"id\",\"0\")],Node [Attribute (\"id\",\"1\")]] [Edge [Attribute (\"source\",\"0\"),Attribute (\"target\",\"1\")]]",
-     P.makePangraph [P.makeNode [P.makeAttribute ("id","0")],P.makeNode [P.makeAttribute ("id","1")]] [P.makeEdge [P.makeAttribute ("source","0"),P.makeAttribute ("target","1")]])
+testShowInstance = do
+  literal <-fmap (head . lines) $ readFile "test/show-string.txt"
```

Comment:
Is the avoidance of IO to make it clear exactly what is being tested? I think maybe the test directory should mirror the src directory in terms of testing this will allow the files to be clearer and do less IO.